filesystem/iso9660.ksy

```diff
@@ -1,210 +1,344 @@
-  speed).
+  endian: be
+doc-ref: |
+  ecma-119 http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-119.pdf
+  susp http://www.ymi.com/ymi/sites/default/files/pdf/Systems%20Use%20P1281.pdf
+  rrip http://www.ymi.com/ymi/sites/default/files/pdf/Rockridge.pdf
+  rras http://www.estamos.de/makecd/Rock_Ridge_Amiga_Specific
+  rrzf https://dev.lovelyhq.com/libburnia/web/wikis/Zisofs
+enums:
+  descriptor_type:
+    0x00: boot_record_volume_descriptor
+    0x01: primary_volume_descriptor
+    0x02: supplementary_volume_descriptor
+    0x03: volume_partition_descriptor
+    0xff: volume_descriptor_set_terminator
+  su_signature:
+    0x4153: rras_amiga_specific # AS
+    0x4345: susp_continuation_area # CE
+    0x434c: rrip_child_link # CL
+    0x4552: susp_extensions_reference # ER
+    0x4553: susp_extension_selector # ES
+    0x4e4d: rrip_alternate_name # NM
+    0x5044: susp_padding_field # PD
+    0x504c: rrip_parent_link # PL
+    0x504e: rrip_posix_device_number # PN
+    0x5058: rrip_posix_file_attributes # PX
+    0x5245: rrip_relocated_directory # RE
+    0x5252: rrip_extensions_in_use_indicator # RR
+    0x5346: rrip_sparse_file # SF
+    0x534c: rrip_symbolic_link # SL
+    0x5350: susp_indicator # SP
+    0x5354: susp_terminator # ST
+    0x5446: rrip_time_file # TF
+    0x5a46: rrzf_zisofs # ZF
 types:
-  vol_desc:
+  u2bi:
+    doc-ref: ecma-119 7.2.3
     seq:
-      - id: type
+      - id: le
+        type: u2le
+      - id: be
+        type: u2be
+  u4bi:
+    doc-ref: ecma-119 7.3.3
+    seq:
+      - id: le
+        type: u4le
+      - id: be
+        type: u4be
+  volume_descriptor:
+    doc-ref: ecma-119 8.1
+    seq:
+      - id: descriptor_type
+        doc-ref: ecma-119 8.1.1
         type: u1
+        enum: descriptor_type
       - id: magic
-        contents: "CD001"
+        doc-ref: ecma-119 8.1.2
+        contents: [0x43, 0x44, 0x30, 0x30, 0x31]
       - id: version
-        type: u1
-      - id: vol_desc_boot_record
-        type: vol_desc_boot_record
-        if: type == 0
-      - id: vol_desc_primary
-        type: vol_desc_primary
-        if: type == 1
-  vol_desc_boot_record:
+        doc-ref: ecma-119 8.1.3
+        contents: [0x01]
+      - id: boot_record
+        type: boot_record_volume
+        if: descriptor_type == descriptor_type::boot_record_volume_descriptor
+      - id: primary_volume
+        type: primary_volume
+        if: descriptor_type == descriptor_type::primary_volume_descriptor
+      - id: supplementary_volume
+        type: supplementary_volume
+        if: descriptor_type == descriptor_type::supplementary_volume_descriptor
+#      - id: volume_partition
+#        type: volume_partition
+#        if: descriptor_type == descriptor_type::volume_partition_descriptor
+  boot_record_volume:
+    doc-ref: ecma-119 8.2
+    seq:
+      - id: boot_system_identifier
+        doc-ref: ecma-119 8.2.4
+        type: strz
+        size: 0x20
+        encoding: ASCII
```

Comment:
```encoding: ASCII``` is so widespread here.

As UTF-8 is superset of ASCII, why not to just put ```encoding: UTF-8``` into ```meta```?

Reply:
I have read a bit, it really doesn't accept utf-8. But you can move ascii to meta too.

Reply:
I can do this, but once I add Juliet then UTF-8 like will be added all over the place.

Reply:
according to wikipedia, joilet uses UCS-2

Reply:
ASCII is now global